midp/gfx.js

```diff
@@ -98,10 +169,20 @@ var currentlyFocusedTextEditor;
 
     var refreshStr = "refresh";
     Native["com/sun/midp/lcdui/DisplayDevice.refresh0.(IIIIII)V"] = function(hardwareId, displayId, x1, y1, x2, y2) {
-        x1 = Math.max(0, x1);
-        y1 = Math.max(0, y1);
-        x2 = Math.max(0, x2);
-        y2 = Math.max(0, y2);
+        if (!config.useOffscreenCanvas) {
+            // If the offscreen canvas is disabled, then we draw directly
+            // to the device canvas, so it's always up-to-date and never needs
+            // to be refreshed.
+            return;
```

Comment:
Are we intending to disable the offscreen canvas for real MIDlets?

In the `DisplayDevice.refresh0` function we use `requestAnimationFrame` to ensure that we draw to the device canvas at an appropriate time. When the offscreen canvas is disabled, we draw to the device canvas directly and we don't take advantage of `requestAnimationFrame` at all.

If we're intending to disable the offscreen canvas for some MIDlets, we should file a follow-up issue to ensure that we only paint to the screen in a callback to `requestAnimationFrame`. We'll maybe have to create some way of batching paint requests.